openlibrary/plugins/openlibrary/js/ile/utils/SelectionManager/SelectionManager.js

```diff
@@ -82,19 +84,84 @@ export default class SelectionManager {
     /**
      * @param {MouseEvent & { currentTarget: HTMLElement }} clickEvent
      */
-    toggleSelected(clickEvent) {
+    processClick(clickEvent) {
         // If there is text selection or the click is on a link that isn't a select handle, don't do anything
-        if (window.getSelection()?.toString() !== '' ||
+        if ((!clickEvent.shiftKey && window.getSelection()?.toString() !== '') ||
             ($(clickEvent.target).closest('a').is('a') &&
             $(clickEvent.target).not('.ile-select-handle').length > 0)) return;
 
         const el = clickEvent.currentTarget;
+        if (clickEvent.shiftKey && this.lastClicked)
+        {
+            // clear selection ranges created by shift-clicking since they're not suppressed by preventDefault().
+            clearTextSelection();
+            const siblingSet = this.getSelectableRange(el);
+            const lastClickedIndex = siblingSet.index(this.lastClicked);
+            const elIndex = siblingSet.index(el);
+            if (lastClickedIndex > -1 && Math.abs(elIndex - lastClickedIndex) > 1) {
+                let affectedElements;
+                if (elIndex > lastClickedIndex) {
+                    affectedElements = siblingSet.slice(lastClickedIndex + 1, elIndex + 1);
+                } else {
+                    affectedElements = siblingSet.slice(elIndex, lastClickedIndex);
+                }
+                const stateChange = this.lastClicked.classList.contains('ile-selected') ? true : false;
+                for (const element of affectedElements) this.toggleSelected(element, stateChange);
```

Comment:
```suggestion
                // Select the affected elements if any are unselected
                const select = affectedElements.some(el => !el.classList.contains('ile-selected'));
                for (const element of affectedElements) this.toggleSelected(element, select);
```

Reply:
This suggestion seems to be based on the assumption that we're only selecting ranges, but my solution implements deselecting a range with shift+click as well, which isn't something that Google Photos does. 

I think you're trying to do something that's already implemented, only in toggleSelected rather than here.